### ISSUE TITLE
non-physical electrons behavior in laser-solid interaction

### ISSUE THREAD

**Issue by aeriforme** [ECP-WarpX/WarpX#1518]
Dear developers, 

I have recently encountered some problems when running 2D simulations with WarpX. The simulations consist of a high-intensity (a0=40) ultra-short (30 fs) laser that interacts with a double-layer solid target of micrometric thickness. The first layer has a density close to the critical value, the second one is a high-density layer. We want to simulate the heating of electrons within the target right after the reflection at the high-density layer.

The problem concerns (at least) the high-energy electrons. In the following plot you can see the electrons with energy above 5 MeV scattered in the (x,y) plane (I use y instead of z). In red are the electrons from the near-critical layer, in purple the ones from the solid-density layer. The shade of the color represents the energy of the particles: the whiter, the more energetic. 

![eles_dots_06228](https://user-images.githubusercontent.com/17280419/99563344-7307fa00-29c9-11eb-93ce-20177e8a748b.png)

As you can see, for some reason the electrons from the back-side layer seem to disappear when they cross the rear target-vacuum interface. Moreover, the electron density plots also show some strange features. Indeed, when plotting the electron density separately for the two layers, it seems like one is reminiscent of the other, as shown in following plots. I also tried to play around with the simulated configuration but I always get analogous results. 
 
![rho_ele_foam_06228](https://user-images.githubusercontent.com/17280419/99562822-db0a1080-29c8-11eb-87d9-523c615f7f8c.png)
![rho_ele_subs_06228](https://user-images.githubusercontent.com/17280419/99562838-de9d9780-29c8-11eb-8927-7c0ee19edcab.png)

This simulation was performed on Marconi100 supercomputer using 2 compute nodes (i.e. 8 CPUs+ 8 GPUs). Here I attach the input file, the job that I used to launch the simulation and the standard output. 

[input.txt](https://github.com/ECP-WarpX/WarpX/files/5561635/input.txt)
[job.txt](https://github.com/ECP-WarpX/WarpX/files/5561638/job.txt)
[output.txt](https://github.com/ECP-WarpX/WarpX/files/5561636/output.txt)

The code version is: 
WarpX: 20.11-14-g70bfde2d9d0e
PICSAR: b78ba49a4299
AMReX: 20.11-36-ga01697fda4a2

I guess I may be making a wrong use of the code, but I cannot identify my mistakes. Do you have any suggestion on how to fix this problem or explanation of why this occurs? 

Cheers and thank you.

**Comment by lucafedeli88** [comment]
Thanks for having opened this issue. We will have a look as soon as possible.

**Comment by ax3l** [comment]
@aeriforme thank you for reporting & posting the details! We are over here coming out of conference weeks and going into the thanksgiving week, sorry for the delay.

Can you please do a plot of the accumulated "rho" to double-check if the overall initial target is as planned quasi-neutral?
It looks to me like the substrate might not be quasi-neutral:
```
ion_subs.charge = 9.613059724800001e-19
```
I think this might have to read `2.0826e-18` for Z=13? (Please double-check `ion_subs.mass`, too.)

Does this help?

Few things I see on the side: feel free to use 3rd order shapes and `warpx.use_filter = 1` for improved numerics.

**Comment by lucafedeli88** [comment]
I can confirm that the charge of the substrate ions is an issue. I ran the simulation myself and I added the densities, obtaining this:
![immagine](https://user-images.githubusercontent.com/4558310/100031203-4d2d8b80-2df5-11eb-9b4c-46e39934f06e.png)
Since at t=0 the fields are zero, this means that the target behaves as if there was an immobile, very dense, ion layer. Such layer can lead to an unphysical confinement effect on the target's electrons.
